### ISSUE TITLE
Api v 2

### ISSUE THREAD

**Issue by Hellowlol** [Tautulli/Tautulli#376]
Atm there are a lot of functions and classes that does aproxomently the same thing and I would like to lessen the burden for the devs.

I have started playing with a new api. The general idea is simply add a decorator to a function and the api will automatically support that call.

The problem is there are a lot of undocumented code and while the name suggest what it's used for the args and kwargs does not. And this has to improve if it should be any point. 

So in short adding: @addtoapi all args, kWs from the user is passed to the function. The response type is evauled and converted to the users wanted output (defaulting to json if omitted)

It this sounds interesting I'll keep working on it.

**Comment by JonnyWong16** [comment]
Yeah, let me know which code and I'll try to add some documentation to it.

**Comment by Hellowlol** [comment]
I was thinking about something like this https://repl.it/B8v7
